### PR TITLE
Add option to use custom shell

### PR DESCRIPTION
**Pull request by albinekb**
By default `shell-env` uses the login shell to get the environment variables from, this PR adds an optional argument that can be used to override that with a shell of choice.

We need this over at [hyperterm](https://github.com/zeit/hyperterm)

Also, i updated it to use `execa.sync` as stated by the TODO ⭐ 

Should it throw an error if `shell` is passed and can't be executed? 
Instead of silently returning process.env..

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,25 +16,24 @@ function parseEnv(env) {
...
+module.exports.sync = shell => {
 	if (process.platform === 'win32') {
 		return process.env;
 	}
 
 	try {
-		// TODO: use `execa` → https://github.com/sindresorhus/execa/issues/7
-		const stdout = childProcess.execFileSync(defaultShell, args, {encoding: 'utf8'});
-		return parseEnv(stdout.trim());
+		const stdout = execa.sync(shell || defaultShell, args);
```
Comment (sindresorhus) [inline index.js]: Should be:

``` js
const stdout = execa.sync(shell || defaultShell, args).stdout;
```

**Comment by sindresorhus** [comment]
> Should it throw an error if shell is passed and can't be executed? 

Yes

File: test.js [hunk 2]
```diff
@@ -12,3 +12,25 @@ test('sync', t => {
...
+	t.true('HOME' in env);
+	t.false('' in env);
+});
+
+test('sync with custom shell throws on non-executable', async t => {
+	t.throws(() => m.sync('non-executable'));
+});
+
+test('async with custom shell throws on non-executable', async t => {
+	t.throws(m('non-executable'));
```
Comment (LinusU) [inline test.js:35]: `await t.throws(m('non-e...`?

**Comment by albinekb** [comment]
Hey @sindresorhus could you take a look? 

I've added tests for errors 🐱

**Comment by sindresorhus** [comment]
Hah, I was actually looking at it right this second.

File: test.js [hunk 3]
```diff
@@ -26,3 +26,11 @@ test('sync with custom shell', t => {
 	t.true('HOME' in env);
 	t.false('' in env);
 });
+
+test('sync with custom shell throws on non-executable', async t => {
```
Comment (sindresorhus) [inline test.js]: Shouldn't be `async`

File: test.js [hunk 4]
```diff
@@ -26,3 +26,11 @@ test('sync with custom shell', t => {
...
+	t.throws(() => m.sync('non-executable'));
+});
+
+test('async with custom shell throws on non-executable', async t => {
+	await t.throws(m('non-executable'));
```
Comment (sindresorhus) [inline test.js]: You don't need to `await` it. AVA handles it for us. Also no need for it to be an `async` function then.

**Comment by albinekb** [comment]
Another option would be to just expose the `parseEnv` function, because this is probably an edge-case?

File: test.js (hunk 4 above)
Comment (LinusU) [inline test.js]: How does that work? Isn't that only if you `return` it since `t.throws` needs to wait for the promise to settle before checking if it rejected or not?

**Comment by sindresorhus** [comment]
Nah, I think this is good.

File: test.js (hunk 4 above)
Comment (sindresorhus) [inline test.js]: No, you only have to `await`/return it if you want the error object. Otherwise AVA handles waiting for the promise to resolve.

``` js
import test from 'ava';

test(t => {
    t.throws(new Promise((resolve, reject) => {
        setTimeout(() => reject(new Error('foo')), 500);
    }));
});
```

Comment (LinusU) [inline test.js]: Wow, ✨ magic ✨

I guess that it's advisable to `await` it if you have stuff happening after though? Since otherwise the code path will continue even if there was an error at that point...

**Comment by albinekb** [comment]
Tests are updated with regards to your comments 🎉

**Comment by sindresorhus** [comment]
Thank you! :)
